### PR TITLE
Allow setting custom match name

### LINKED ISSUE #28: Allow setting custom name for game room
- [ ] Add optional custom name input field for created game room.
- [ ] Add search bar at the top of game Lobby to search by name or connected players

### LINKED ISSUE #121: Remember last game
In the lobby, when creating new room our last played game should be preselected in form

### PR DESCRIPTION
**Pull request by pastpatryk**
This PR adds ability to set (or generate) custom match name for the newly created game match.
It also saves previously created game settings and restores them on every page refresh.

image

Closes #28
Closes #121

### CODE REVIEW

File: src/components/layout/GameLayout.js [hunk 1]
```diff
@@ -33,9 +34,9 @@ const defaultProps = {
 const MainMenu = React.memo(({ gameName, privateMatch }) => (
   <div className={styles.mainMenu}>
     <Container>
-      <span className={styles.logo}>
+      <Link className={styles.logo} to="/">
```
Comment (glowacki-dev) [inline src/components/layout/GameLayout.js:37]: Haha this thing not being a link to lobby was annoying 😄 That's a good change

File: src/utils/generators/locales/en.json [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+    "bully kutta",
+    "burgos pointer",
+    "affenpinscher",
+    "afghan hound",
+    "aidi",
+    "airedale terrier",
+    "akbash",
+    "akita",
+    "alano español",
+    "alapaha blue blood bulldog",
```
Comment (glowacki-dev) [inline src/utils/generators/locales/en.json:170]: Some of those get a bit too long 
obraz
I'd stick to simple, short words

File: src/components/lobby/CreateMatchForm.js [hunk 3]
```diff
@@ -61,13 +73,21 @@ const CreateMatchForm = ({ games, onCreate, disabled, loading }) => {
         value={players}
         onChange={(_, { value }) => setPlayers(value)}
       />
+      <Form.Input
+        fluid
+        label={`${t("create.match_name")} (${t("create.optional")})`}
+        value={name}
```
Comment (glowacki-dev) [inline src/components/lobby/CreateMatchForm.js:79]: I'd either pre-fill this with a generated name, or generate some name if user left this field blank, so we can avoid having random characters as room names

Comment (pastpatryk) [reply src/components/lobby/CreateMatchForm.js:81]: I left the field empty by default, but made it required, so you need to click generate to fill it.
I think it's better this way, so users know where this name come from.

There is also a prompt message that instructs how to generate name

File: src/utils/generators/names.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import en from "./locales/en.json";
+import pl from "./locales/pl.json";
+import { sample } from "lodash";
+
+const languages = { en, pl };
+
+function conjugateAdj(lang, adjective) {
```
Comment (glowacki-dev) [inline src/utils/generators/names.js:7]: Now it seems like we could get rid of this method completely 😄

**Review by glowacki-dev — APPROVED** [review]

**Pull request merged**